spec/services/set_decline_by_default_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe SetDeclineByDefault do
+  describe '#call' do
+    let(:application_form) { create(:completed_application_form) }
+    let(:choices) { application_form.application_choices }
+    let(:time_limit_calculator) { instance_double('TimeLimitCalculator', call: 10) }
+    let(:now) { Time.zone.local(2019, 11, 26, 12, 0, 0) }
+    let(:call_service) { SetDeclineByDefault.new(application_form: application_form).call }
+
+    before { allow(TimeLimitCalculator).to receive(:new).and_return(time_limit_calculator) }
+
+    around do |example|
+      Timecop.freeze(now) do
+        example.run
+      end
+    end
+
+    def expect_all_decline_by_default_at_values_to_be(expected_value)
+      application_form.application_choices.reload.each do |application_choice|
+        dbd_at = application_choice.decline_by_default_at
+        without_milliseconds = dbd_at && dbd_at.change(usec: 0)
```

Comment:
Doesn't timecop help us avoid having to do this?

Reply:
Weirdly, no. The explicit reload, which makes the application_choices visible to the test, fetch the timestamp from the database and the value from the database seems to have different precision/scale than the memory-based calculated Ruby time. We've tried using ```.round```, but ```:end_of_day``` setting .99999999 means that round rounds up to the following day.